extensions/amp-sidebar/1.0/component.js

```diff
@@ -190,45 +191,51 @@ export function SidebarToolbar({
   children,
   ...rest
 }) {
-  const ref = useRef();
-  const [matches, setMatches] = useState(false);
-  const [targetEl, setTargetEl] = useState(null);
+  const ref = useRef(null);
+  const [target, setTarget] = useState(null);
 
   useEffect(() => {
-    const window = ref.current?.ownerDocument?.defaultView;
-    if (!window) {
+    const doc = ref.current?.ownerDocument;
+    if (!doc) {
       return;
     }
 
-    const mediaQueryList = window.matchMedia(mediaQueryProp);
-    const updateMatches = () => setMatches(mediaQueryList.matches);
-    mediaQueryList.addEventListener('change', updateMatches);
-    setMatches(mediaQueryList.matches);
-    return () => mediaQueryList.removeEventListener('change', updateMatches);
-  }, [mediaQueryProp]);
+    const selector = `#${escapeCssSelectorIdent(toolbarTarget)}`;
+    const newTarget = doc.querySelector(selector);
+    setTarget(newTarget);
+  }, [toolbarTarget]);
 
   useEffect(() => {
-    const document = ref.current?.ownerDocument;
-    if (!document) {
+    const element = ref.current;
+    const doc = ref.current?.ownerDocument;
+    if (!doc || !target) {
       return;
     }
 
-    const selector = `#${CSS.escape(toolbarTarget)}`;
-    const newTargetEl = document.querySelector(selector);
-    setTargetEl(newTargetEl);
-  }, [toolbarTarget]);
+    const clone = element.cloneNode(true);
```

Comment:
Note that this can be run frequently if the target or mediaQuery change, which can be expensive.